### PR TITLE
Fixes #323 - Support editing components and sub-components

### LINKED ISSUE #323: Batch metadata overlay: deliverable 3
### Descriptive summary
Component metadata can be edited

### Rationale
We should be able to edit exiting component metadata.

[Batch edit specification](https://docs.google.com/document/d/17e921jvvjhFXJA_y1VaDyi9Qmsg_ZNYv4UFSyxACQDM/edit#)

### Related work
#321
#322
#324

### PR DESCRIPTION
**Pull request by lsitu**
Fixes #323

#### Local Checklist
- [x] Tests written and passing locally?
- [x] Code style checked?
- [x] QA-ed locally?
- [x] Rebased with `master` branch?
- [ ] Configuration updated (if needed)?
- [ ] Documentation updated (if needed)?

#### What does this PR do?
Support editing components and sub-components for batch metadata overlay.

##### Why are we doing this? Any context of related work?
References #321 #322 #323 #324

#### Screenshots

---

#### New ENV variables
References #issuenumber

#### Deployment Instructions

@ucsdlib/developers - please review

### CODE REVIEW

File: test/edu/ucsd/library/xdre/tab/BatchEditExcelSourceTest.java [hunk 1]
```diff
@@ -30,6 +30,7 @@ public void init() throws Exception {
         Constants.DAMS_STORAGE_URL = "http://localhost:8080/dams/api";
         Constants.DAMS_ARK_URL_BASE = "http://library.ucsd.edu/ark:";
         Constants.ARK_ORG = "20775";
+        Constants.BATCH_ADDITIONAL_FIELDS = "Note:local attribution,collection(s)";
```
Comment (mcritchlow) [inline test/edu/ucsd/library/xdre/tab/BatchEditExcelSourceTest.java]: I'm curious how these 'additonal fields' are used. This is not a concern or anything, I'm just curious what their purpose is.

File: test/edu/ucsd/library/xdre/tab/BatchEditExcelSourceTest.java [hunk 2]
```diff
@@ -157,5 +158,17 @@ public void testNextRecord() throws Exception {
...
         // collection(s)
         assertEquals("collection(s) doesn't match!", "Collection(s)", data.get("collection(s)"));
+
+        // Component
+        List<TabularRecord> comps = editRecord.getComponents();
+        assertEquals("component size doesn't match!", 1, comps.size());
+        Map<String, String> compData = comps.get(0).getData();
+        assertEquals("component title doesn't match!", "Test Component 1", compData.get("title"));
+
+        // Sub-Component
```
Comment (mcritchlow) [inline test/edu/ucsd/library/xdre/tab/BatchEditExcelSourceTest.java]: This is entirely up to you, but a question I have is should we have some kind of test/validation that ensures no one tries to accidentally(or intentionally..) do a sub-sub Component? Basically test/verify that we're only supporting a sub-component level of depth? I think you would know better than me if that's a legitimate concern for your implementation. Just a question that popped into my mind while reviewing these tests.

File: test/edu/ucsd/library/xdre/tab/BatchEditExcelSourceTest.java (hunk 1 above)
Comment (lsitu) [reply test/edu/ucsd/library/xdre/tab/BatchEditExcelSourceTest.java]: My thoughts for the additional fields are the fields that need validation in Batch Overlay but not Excel InputStream, which could be added by default, and I am not sure whether how it'll be changed while moving forward with the support with decoupling rights from the Excel InputStream form.

File: test/edu/ucsd/library/xdre/tab/BatchEditExcelSourceTest.java (hunk 2 above)
Comment (lsitu) [reply test/edu/ucsd/library/xdre/tab/BatchEditExcelSourceTest.java]: Good question. I don't think it'll work with deeper level of depth with the logic and implementation we have for the Excel InputStream: rows of `Component` and `Sub-component`. It'll be always a new Component OR a Sub-component that under the current component context.

File: test/edu/ucsd/library/xdre/tab/BatchEditExcelSourceTest.java (hunk 1 above)
Comment (mcritchlow) [reply test/edu/ucsd/library/xdre/tab/BatchEditExcelSourceTest.java]: Ah, that makes sense. Thank you!

File: test/edu/ucsd/library/xdre/tab/BatchEditExcelSourceTest.java (hunk 2 above)
Comment (mcritchlow) [reply test/edu/ucsd/library/xdre/tab/BatchEditExcelSourceTest.java]: Ok sounds good. Thanks.

**Pull request merged**